core/src/main/java/org/apache/iceberg/avro/TypeToSchema.java

```diff
@@ -232,4 +235,32 @@ public Schema primitive(Type.PrimitiveType primitive) {
 
     return primitiveSchema;
   }
+
+  // This function ensures that all nested null are converted to JsonProperties.NULL_VALUE
+  // to make sure JacksonUtils.toJsonNode() converts them properly.
+  private Object convertToJsonNull(Object defaultValue) {
+    if (defaultValue instanceof Map) {
+      for (Map.Entry<Object, Object> entry : ((Map<Object, Object>) defaultValue).entrySet()) {
+        if (entry.getValue() instanceof Map || entry.getValue() instanceof Collection) {
+          entry.setValue(convertToJsonNull(entry.getValue()));
+        } else {
```

Comment:
wouldn't you need a condition sth like `else if (entry.getValue() == null)`? same for line 258

Reply:
This change should be in-place, so if there is no change needed, we don't have to check that branch.